client/app/scripts/charts/nodes-layout.js

```diff
@@ -402,7 +402,7 @@ function cloneLayout(layout, nodes, edges) {
 function copyLayoutProperties(layout, nodeCache, edgeCache) {
   const result = Object.assign({}, layout);
   result.nodes = layout.nodes.map(node => (nodeCache.has(node.get('id'))
-    ? node.merge(nodeCache.get(node.get('id'))) : node));
+    ? nodeCache.get(node.get('id')).merge(node) : node));
```

Comment:
I'd like to play with this a bit to see if it really works as it should. I think the idea here was that the cached `x`, `y` values should override the current ones. If `node` object doesn't have them set, then this fix should work just fine. Otherwise, the old direction should be kept for `x`, `y` field, while this direction should be used for all the other node data.

Maybe a more robust solution would be to cache *only* `x`, `y` fields for the nodes (since they are the only fields relevant for the layouts + maybe adjacencies), i.e. to replace

```
cache.nodeCache = cache.nodeCache.merge(layout.nodes)
```

with something like

```
cache.nodeCache = cache.nodeCache.merge(layout.nodes.map(n => fromJS({ x: n.get('x'), y: n.get('y') }))
```